Adds th'oom

JoeyNosegay opened the pull request:
🆑 JoeyNosegay
rscadd: Adds Th'oom simplemob
/🆑

Adds the Th'oom, a massive, domestic reptile native to Qerr'Balak. It can be tipped like a cow, can have it's glands expunged for nutrient, and contains a large amount of valuable reptile steaks. 

![image](https://user-images.githubusercontent.com/21371500/190838656-8405e20c-4018-4369-880f-00caf2025bcf.png)

Doesn't add any spawning anywhere, just adds the mob for admins to cause trouble with.

SierraKomodo commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,69 @@ var/global/chicken_count = 0
+
+	var/datum/reagents/udder = null
+
+	ai_holder = /datum/ai_holder/simple_animal/passive/
```
```suggestion
	ai_holder = /datum/ai_holder/simple_animal/passive
```

SierraKomodo commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,69 @@ var/global/chicken_count = 0
+	udder = new(50, src)
+	..()
+
+/mob/living/simple_animal/passive/thoom/attackby(obj/item/O as obj, mob/user as mob)
```
```suggestion
/mob/living/simple_animal/passive/thoom/attackby(obj/item/O, mob/user)
```

SierraKomodo commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,69 @@ var/global/chicken_count = 0
+/mob/living/simple_animal/passive/thoom/attackby(obj/item/O as obj, mob/user as mob)
+	var/obj/item/reagent_containers/glass/G = O
+	if(stat == CONSCIOUS && istype(G) && G.is_open_container())
+		user.visible_message("<span class='notice'>[user] expunges [src] using \the [O].</span>")
```
```suggestion
		user.visible_message(
			SPAN_NOTICE("\The [user] expunges \the [src] using \the [O]."),
			SPAN_NOTICE("You expunge \the [src] using \the [O].")
		)
```

SierraKomodo replied:
Also, this message should probably only appear after the volume and transferred checks - Otherwise, the displayed message isn't entirely accurate and the user gets two messages (Both the failure and the success message).

SierraKomodo commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,69 @@ var/global/chicken_count = 0
+		user.visible_message("<span class='notice'>[user] expunges [src] using \the [O].</span>")
+		var/transfered = udder.trans_type_to(G, /datum/reagent/toxin/fertilizer/eznutrient, rand(10,15))
+		if(G.reagents.total_volume >= G.volume)
+			to_chat(user, "<span class='warning'>\The [O] is full.</span>")
```
```suggestion
			to_chat(user, SPAN_WARNING("\The [O] is full."))
```

SierraKomodo commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,69 @@ var/global/chicken_count = 0
+		if(G.reagents.total_volume >= G.volume)
+			to_chat(user, "<span class='warning'>\The [O] is full.</span>")
+		if(!transfered)
+			to_chat(user, "<span class='warning'>The gland is dry. Wait a bit longer...</span>")
```
```suggestion
			to_chat(user, SPAN_WARNING("\The [src]'s gland is dry. Wait a bit longer..."))
```

SierraKomodo commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,69 @@ var/global/chicken_count = 0
+	if(stat == CONSCIOUS && istype(G) && G.is_open_container())
+		user.visible_message("<span class='notice'>[user] expunges [src] using \the [O].</span>")
+		var/transfered = udder.trans_type_to(G, /datum/reagent/toxin/fertilizer/eznutrient, rand(10,15))
+		if(G.reagents.total_volume >= G.volume)
+			to_chat(user, "<span class='warning'>\The [O] is full.</span>")
+		if(!transfered)
+			to_chat(user, "<span class='warning'>The gland is dry. Wait a bit longer...</span>")
+	else
+		..()
```
You're doing an `if - if - else` chain here which means that even if the first if statement passes and displays the 'O is full' message, the following if and else will still run, which means you may have situations of `'The bucket is full'` and then it tries to fill the bucket anyway and wastes reagent contents. The fix for that would be to switch to an `if - elseif - else` chain or to just `return` after each condition that should stop due to an error or other failure.

SierraKomodo commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,69 @@ var/global/chicken_count = 0
+	if(udder && prob(5))
+		udder.add_reagent(/datum/reagent/toxin/fertilizer/eznutrient, rand(5, 10))
+
+/mob/living/simple_animal/passive/thoom/attack_hand(mob/living/carbon/M as mob)
```
```suggestion
/mob/living/simple_animal/passive/thoom/attack_hand(mob/living/carbon/M)
```

SierraKomodo commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,69 @@ var/global/chicken_count = 0
+
+/mob/living/simple_animal/passive/thoom/attack_hand(mob/living/carbon/M as mob)
+	if(!stat && M.a_intent == I_DISARM && icon_state != icon_dead)
+		M.visible_message("<span class='warning'>[M] tips over [src].</span>","<span class='notice'>You tip over [src].</span>")
```
```suggestion
		M.visible_message(
			SPAN_WARNING("\The [M] tips over \the [src].",
			SPAN_NOTICE("You tip over \the [src].")
		)
```

SierraKomodo commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,69 @@ var/global/chicken_count = 0
+		M.visible_message("<span class='warning'>[M] tips over [src].</span>","<span class='notice'>You tip over [src].</span>")
+		Weaken(30)
+		icon_state = icon_dead
+		spawn(rand(20,50))
+			if(!stat && M)
+				icon_state = icon_living
+				var/list/responses = list(	"[src] looks at you imploringly.",
+											"[src] looks at you pleadingly",
+											"[src] looks at you with a resigned expression.",
+											"[src] seems resigned to its fate.")
+				to_chat(M, pick(responses))
```
No spawns in new code if possible. Replace this with a timer.

SierraKomodo replied:
```suggestion
				var/list/responses = list(	"looks at you imploringly.",
											"looks at you pleadingly",
											"looks at you with a resigned expression.",
											"seems resigned to its fate.")
				to_chat(M, "\The [src] [pick(responses)]")
```

Spookerton commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,75 @@ var/global/chicken_count = 0
+/mob/living/simple_animal/passive/thoom/attack_hand(mob/living/carbon/M)
+	if(!stat && M.a_intent == I_DISARM && icon_state != icon_dead)
+		M.visible_message(
+			SPAN_WARNING("\The [M] tips over \the [src].",
```
```suggestion
			SPAN_WARNING("\The [M] tips over \the [src]."),
```

Spookerton commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -321,3 +321,75 @@ var/global/chicken_count = 0
+		spawn(rand(20,50))
+			if(!stat && M)
+				icon_state = icon_living
+				var/list/responses = list(	"looks at you imploringly.",
+											"looks at you pleadingly",
+											"looks at you with a resigned expression.",
+											"seems resigned to its fate.")
```
```suggestion
				var/list/responses = list(
					"looks at you imploringly.",
					"looks at you pleadingly",
					"looks at you with a resigned expression.",
					"seems resigned to its fate."
				)
```

Spookerton reviewed (dismissed):
scronch pls

SierraKomodo reviewed (dismissed):
Also due to some repo updates, please rebase on the latest `/dev` and push.

JoeyNosegay committed:
Adds th'oom


Update code/modules/mob/living/simple_animal/friendly/farm_animals.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/simple_animal/friendly/farm_animals.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/simple_animal/friendly/farm_animals.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/simple_animal/friendly/farm_animals.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/simple_animal/friendly/farm_animals.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/simple_animal/friendly/farm_animals.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/simple_animal/friendly/farm_animals.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/simple_animal/friendly/farm_animals.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/simple_animal/friendly/farm_animals.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/mob/living/simple_animal/friendly/farm_animals.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Adds thoom 2
